### PR TITLE
Use default.qubit.jax instead of default.qubit

### PR DESCRIPTION
**Pull request by trbromley**
Ensures users can execute with finite shots (though differentiation is not supported)

### CODE REVIEW

File: doc/releases/changelog-0.29.0.md [hunk 1]
```diff
@@ -52,11 +52,11 @@
...
 
-  Pulse-based circuits can be executed and differentiated on the `default.qubit` simulator using JAX
-  as an interface:
+  Pulse-based circuits can be executed and differentiated on the `default.qubit.jax` simulator using
+  JAX as an interface:
 
   ```pycon
-  >>> dev = qml.device("default.qubit", wires=2)
+  >>> dev = qml.device("default.qubit.jax", wires=2)
   >>> qnode = qml.QNode(pulse_circuit, dev, interface="jax")
```
Comment (Qottmann) [inline doc/releases/changelog-0.29.0.md:60]:
```suggestion
  >>> qnode = qml.QNode(pulse_circuit, dev)
```

File: pennylane/ops/functions/evolve.py [hunk 2]
```diff
@@ -138,7 +138,7 @@ def evolve(*args, **kwargs):  # pylint: disable=unused-argument
 
         import jax
 
-        dev = qml.device("default.qubit", wires=4)
+        dev = qml.device("default.qubit.jax", wires=4)
         @jax.jit
         @qml.qnode(dev, interface="jax")
```
Comment (Qottmann) [inline pennylane/ops/functions/evolve.py:143]:
```suggestion
        @qml.qnode(dev)
```

File: pennylane/pulse/__init__.py [hunk 3]
```diff
@@ -212,7 +212,7 @@
 
     import jax
 
-    dev = qml.device("default.qubit", wires=1)
+    dev = qml.device("default.qubit.jax", wires=1)
 
     @jax.jit
     @qml.qnode(dev, interface="jax")
```
Comment (Qottmann) [inline pennylane/pulse/__init__.py:218]:
```suggestion
    @qml.qnode(dev)
```

File: pennylane/pulse/convenience_functions.py [hunk 4]
```diff
@@ -57,7 +57,7 @@ def constant(scalar, time):
 
     .. code-block:: python
 
-        dev = qml.device("default.qubit", wires=1)
+        dev = qml.device("default.qubit.jax", wires=1)
         @qml.qnode(dev, interface="jax")
```
Comment (Qottmann) [inline pennylane/pulse/convenience_functions.py:61]:
```suggestion
        @qml.qnode(dev)
```

File: pennylane/pulse/parametrized_evolution.py [hunk 5]
```diff
@@ -114,7 +114,7 @@ class ParametrizedEvolution(Operation):
 
         import jax
 
-        dev = qml.device("default.qubit", wires=1)
+        dev = qml.device("default.qubit.jax", wires=1)
         @jax.jit
         @qml.qnode(dev, interface="jax")
```
Comment (Qottmann) [inline pennylane/pulse/parametrized_evolution.py:119]:
```suggestion
        @qml.qnode(dev)
```

File: pennylane/pulse/parametrized_evolution.py [hunk 6]
```diff
@@ -195,7 +195,7 @@ def f2(p, t):
 
         .. code-block:: python
 
-            dev = qml.device("default.qubit", wires=3)
+            dev = qml.device("default.qubit.jax", wires=3)
 
             @qml.qnode(dev, interface="jax")
```
Comment (Qottmann) [inline pennylane/pulse/parametrized_evolution.py:200]:
```suggestion
            @qml.qnode(dev)
```

**Review by Qottmann — COMMENTED** [review]
Thanks for catch @lillian542 and setting this up @trbromley 

Proposing to use _just_ use `default.qubit.jax` and not additionally specifying the interface in the qnode.

**Review by Qottmann — APPROVED** [review]

**Review by AlbertMitjans — COMMENTED** [review]
It would be nice to change the error message in `DefaultQubit._apply_parametrized_evolution` to say: `"use the default.qubit.jax device"`

**Comment by Qottmann** [comment]
Reverted my suggestions to remove `interface="jax"` as this prohibits using non-jax-array params (i.e. lists and tuples which we actively promote) in a jit pipeline:

```python
dev = qml.device("default.qubit.jax", range(4))

ts = jnp.array([0.0, 3.0])

Ht = jnp.polyval * qml.PauliX(0) + jnp.polyval * qml.PauliY(0)

params = [jnp.ones(2), jnp.ones(3)]

@jax.jit
@qml.qnode(dev)
def qnode(params):
    qml.evolve(Ht)(params, ts)
    return qml.expval(qml.PauliX(0))


print(qnode(params))
```

```pycon
..
ValueError: Converting a JAX array to a NumPy array not supported when using the JAX JIT.
```

**Pull request merged**